extmod/asyncio/core.py

```diff
@@ -54,7 +55,8 @@ def __next__(self):
 # Use a SingletonGenerator to do it without allocating on the heap
 def sleep_ms(t, sgen=SingletonGenerator()):
     assert sgen.state is None
-    sgen.state = ticks_add(ticks(), max(0, t))
+    now = ticks()
+    sgen.state = ticks_add(now, t) if t > 0 else now
```

Comment:
Does this give a measurable speed improvement?  Is it worth it for the cost in code size?

I measure this change here as +5 bytes to the bytecode.  The most taken path will be when `ticks_add()` needs to be called, which goes from 12 opcodes previously to now 16 opcodes.  It's usually the opcode overhead that's slow, rather than the actual call (eg out to `max`, which should be quick with two small int args).  So I would guess that this change actually makes things a little slower.

Reply:
@dpgeorge thank you for asking. I spent too long working on this. It turns out that it does make a performance improvement on all platforms (ie. both Unix and MCUs), but it isn't substantial. I'm happy to remove it and search for a more substantial impact.

Reply:
Thanks for investigating.  Unless it's a significant improvement, I'd prefer to leave it as-is (ie prefer shorter bytecode over performance).